docs/docs/code/nodejs/README.md

```diff
@@ -200,7 +206,131 @@ You can return HTTP responses from [HTTP-triggered workflows](/workflows/steps/t
 
 ## Managing state
 
-See the docs on [workflow state](/workflows/steps/code/state/).
+In Node.js (Javascript) code steps, you can also store and retrieve data in code steps.
+
+This is very useful for tracking data between runs of a particular workflow.
+
+:::warning
+This functionality (`$.service.db`) is limited to only Node.js code steps at this time.
+
+Other step languages like [Python](/code/python), [bash](/code/bash) and [Go](/code/go) do not have this feature available yet.
+
+For more information on what functionality is available for those languages, please refer to their documentation.
+:::
+
+### Injecting the database
+
+By default Node.js steps do not contain the database service. It needs to be injected by defining it as a `prop`. 
+
+```javascript
+defineComponent({
+  props: {
+    // Define that the "db" variable in our component is a database
+    db: "$.service.db",
+  },
+  async run({ steps, $ }) {
+    // Now we can access the database at "this.db"
+    this.db
+  })
+});
+```
+
+:::tip
+`props` injects variables under `this` scope in components.
+
+In the above example we essentially instructed that this step needs the database injected into the `this.db` prop. 
+:::
+
+### Using the database
+
+Once the database is injected into the component, we can use it to both store & retrieve data generated from workflow steps. The 2 functions available are to `set` and `get` the data.
```

Comment:
"store **and** retrieve"

Reply:
You can combine these two sentences to shorten, maybe something like this:

> Once you inject the database into the component, you can use it to both store (`set`) and retrieve (`get`) data.

Reply:
Nice, I like it.